crates/router/src/core/subscription.rs

```diff
@@ -63,3 +68,549 @@ pub async fn create_subscription(
+                        message: "item_price_id is required".to_string(),
+                    },
+                )?,
+                quantity: Some(1),
+            };
+        let subscription_req = hyperswitch_domain_models::router_request_types::subscriptions::SubscriptionCreateRequest {
+            subscription_id: self.subscription.id.to_owned(),
+            customer_id: self.subscription.customer_id.to_owned(),
+            subscription_items: vec![subscription_item],
+            billing_address: self.request.billing_address.clone().ok_or(errors::ApiErrorResponse::MissingRequiredField {
+                field_name: "billing_address",
+            })?,
+            auto_collection: hyperswitch_domain_models::router_request_types::subscriptions::SubscriptionAutoCollection::Off,
+            connector_params: self.connector_params.clone(),
+         };
+
+        let router_data = self.build_router_data(state, subscription_req)?;
+        let connector_integration = self.connector_data.connector.get_connector_integration();
+
+        let response = self
+            .call_connector(
+                state,
+                router_data,
+                "create subscription",
+                connector_integration,
+            )
+            .await?;
+
+        match response {
+            Ok(response_data) => Ok(response_data),
+            Err(err) => Err(errors::ApiErrorResponse::ExternalConnectorError {
+                code: err.code,
+                message: err.message,
+                connector: self.connector_name.to_string(),
+                status_code: err.status_code,
+                reason: err.reason,
+            }
+            .into()),
+        }
+    }
+
+    async fn call_connector<F, ResourceCommonData, Req, Resp>(
+        &self,
+        state: &SessionState,
+        router_data: hyperswitch_domain_models::router_data::RouterData<F, Req, Resp>,
+        operation_name: &str,
+        connector_integration: hyperswitch_interfaces::connector_integration_interface::BoxedConnectorIntegrationInterface<F, ResourceCommonData, Req, Resp>,
+    ) -> errors::RouterResult<Result<Resp, hyperswitch_domain_models::router_data::ErrorResponse>>
+    where
+        F: Clone + std::fmt::Debug + 'static,
+        Req: Clone + std::fmt::Debug + 'static,
+        Resp: Clone + std::fmt::Debug + 'static,
+        ResourceCommonData:
+            hyperswitch_interfaces::connector_integration_interface::RouterDataConversion<
+                    F,
+                    Req,
+                    Resp,
+                > + Clone
+                + 'static,
+    {
+        let router_resp = services::execute_connector_processing_step(
+            state,
+            connector_integration,
+            &router_data,
+            payments_core::CallConnectorAction::Trigger,
+            None,
+            None,
+        )
+        .await
+        .change_context(errors::ApiErrorResponse::InternalServerError)
```

Comment:
```suggestion
        .change_context(errors::ApiErrorResponse::SubscriptionFailed)
```
Add new APIErrorResponse for subscription failure